sdv/single_table/base.py

```diff
@@ -110,22 +110,22 @@ def __init__(
         self, metadata, enforce_min_max_values=True, enforce_rounding=True, locales=['en_US']
     ):
         self._validate_inputs(enforce_min_max_values, enforce_rounding)
-        self.metadata = metadata
-        self._table_name = Metadata.DEFAULT_SINGLE_TABLE_NAME
         if isinstance(metadata, Metadata):
             self._table_name = metadata._get_single_table_name()
-            self.metadata = metadata._convert_to_single_table()
-        elif isinstance(metadata, SingleTableMetadata):
+            self.metadata = metadata
+        else:
             warnings.warn(DEPRECATION_MSG, FutureWarning)
+            self._table_name = Metadata.DEFAULT_SINGLE_TABLE_NAME
+            self.metadata = Metadata.load_from_dict(metadata.to_dict(), self._table_name)
+            self.metadata.tables[self._table_name]._updated = metadata._updated
```

Comment:
why do we need to set the updated flag here?

Reply:
Loading from the dict in line 119 doesn't set the `_updated` attribute, we need to do it manually.

Reply:
but how was it updated? It just loaded from a dict right? Did any attributes change

Reply:
Updating a single table metadata may set `_updated = True`, like [here](https://github.com/sdv-dev/SDV/blob/391fd824fad082a97207705c522a61b7ae0ca68a/sdv/metadata/single_table.py#L294). One example is [this](https://github.com/sdv-dev/SDV/blob/391fd824fad082a97207705c522a61b7ae0ca68a/tests/integration/single_table/test_base.py#L719) test where the metadata is updated before initializing the synthesizer.